books/free-programming-books-subjects.md

```diff
@@ -112,6 +112,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Essential Algorithms](https://www.programming-books.io/essential/algorithms) - Krzysztof Kowalczyk and  Stack Overflow Documentation project (HTML)
 * [Foundations of Computer Science](http://infolab.stanford.edu/~ullman/focs.html) - Al Aho, Jeff Ullman
 * [Geometry Algorithms](http://geomalgorithms.com) - Dan Sunday
+* [Introduction to Algorithms](https://sd.blackball.lv/library/Introduction_to_Algorithms_Third_Edition_(2009).pdf) - Thomas H. Cormen, Charles E. Leiserson, Ronald L. Rivest, Clifford Stein (PDF)
```

Comment:
Is this an authorized distribution?

Reply:
Hi. Could you please elaborate the question a little bit? Actually, the link to the book is publicly available on the Internet. There are various other links too available for the same.